src/Microsoft.Data.Sqlite.Core/SqliteCommand.cs

```diff
@@ -322,6 +322,11 @@ public new virtual SqliteDataReader ExecuteReader(CommandBehavior behavior)
                     ? PrepareAndEnumerateStatements()
                     : _preparedStatements)
                 {
+                    if (!unprepared)
+                    {
+                        raw.sqlite3_reset(stmt);
```

Comment:
In case of INSERTs (as in the test case) the DataReader has no access to the statement, since we only read back the value of changed rows (passed queue is null).